tools/format.py

```diff
@@ -20,14 +41,7 @@ def run(**kwargs):
 
     # Use git to get all of the files that are committed to the repository
     files = check_output(["git", "ls-files"]).decode("utf-8")
-    for f in files.splitlines():
-        if f.endswith((".h", ".c", ".cc", ".cxx", ".cpp", ".hpp", ".ipp", ".proto")):
-            print("Formatting {} with clang-format".format(f))
-            call(["clang-format", "-i", "-style=file", f])
-        elif f.endswith(("CMakeLists.txt", ".cmake", ".role")):
-            print("Formatting {} with cmake-format".format(f))
-            call(["cmake-format", "-i", f])
-        elif f.endswith((".py")):
-            print("Formatting {} with isort and black".format(f))
-            call(["isort", f])
-            call(["black", f])
+
+    with multiprocessing.Pool(multiprocessing.cpu_count()) as pool:
+        for r in pool.imap_unordered(_do_format, files.splitlines()):
```

Comment:
I was curious about whether changing the chunk size would have a large impact on the performance, so I unscientifically benched it on my pc and was only able to squeeze out an approx 1.5 s (~10%) run time improvement, with a chunk size of 20.

Reply:
Interesting... I wouldn't have thought it would make much of a difference